### PR TITLE
You can now fish with explosives

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Iansdoor suggested this to me earlier (but I already knew it was a thing on Goon), and I said ok. Basically, this PR adds an interaction between explosions and fishing.

At the cost of potential collateral damage, you can too now use a TTV to catch a lot of fishes (and other stuff). Fish spawned this way will be pretty much dead however.

This PR also adds the examine tips already presents in the fishing spot component to the lazy fishing spot element.

## Why It's Good For The Game
This PR adds an interaction between fishing and other mechanics of the game.

## Changelog

:cl:
add: You can now fish with explosives.
fix: Fixed an inconsistency with examining fishing spots with sufficiently high fishing skill (or skillchip).
/:cl:

### CODE REVIEW

File: code/modules/fishing/sources/_fish_source.dm [hunk 1]
```diff
@@ -53,6 +53,10 @@ GLOBAL_LIST_INIT(specific_fish_icons, zebra_typecacheof(list(
 	var/catalog_description
 	/// Background image name from /datum/asset/simple/fishing_minigame
 	var/background = "background_default"
+	/// It true, repeated and large explosions won't be as efficient. This is usually meant for global fish sources.
+	var/explosive_malus = FALSE
+	/// If explosive_malus is true, this will be used to keep track of the turfs where an explosion happened for when we'll spawn the loot.
+	var/list/exploded_turfs
```
Comment (00-Steven) [inline code/modules/fishing/sources/_fish_source.dm:59]: I was testing this pr, putting gibtonite at the edge of a lavaland lava river and having it explode for fish- I could only do this about once, before I could no longer get fish from the lava with any explosive I tried.
My first thought is, I don't think this ever gets reset anywhere? But I haven't tried to debug much yet.

Comment (Ghommie) [reply code/modules/fishing/sources/_fish_source.dm:59]: OH, YOU'RE RIGHT! I TOTALLY FORGOT!

**Comment by Dawnseer** [comment]
Finally, a less harmful recreational use of pipebombs. Those fish won't know what hit them.

**Comment by 00-Steven** [comment]
o7

**Comment by Dawnseer** [comment]
Please tell me that if I add glass shards to the pipe bombs that I go fishing with, it'l add some extra crunch to the fish

**Comment by Stoopler** [comment]
Can you throw a TTV into a chasm to have it violently eject anything that was in it?

**Comment by Ghommie** [comment]
> Can you throw a TTV into a chasm to have it violently eject anything that was in it?

Alas, no. But that interaction between ttv and chasms can be done in a different PR, without touching fishing code.

**Pull request merged**